File: list_devices.py
#!/usr/bin/python

import pygame.midi, json

pygame.midi.init()
c = pygame.midi.get_count()
devices = "%s midi devices found" % c + "\n"
for i in range(c):
    devices += "%s: " % i
    devices += "%s name: %s input: %s output: %s opened: %s" % (pygame.midi.get_device_info(i))+ "\n"

devices += "Default device: " + str(pygame.midi.get_default_output_id())
print(devices)

File: list_devices.php
<?php
// call python script with appropriate arguments in JSON format
$pyScript = 'list_devices.py';
$pythonExec = '/usr/bin/python';
$cmd = "sudo -u www-data $pythonExec $pyScript";
echo nl2br(shell_exec($cmd));

/*
// query alsa for midi devices 
$output = shell_exec("sudo -u www-data amidi -l");
error_log($output);
4 midi devices found\nALSA name: Midi Through Port-0 input: 0 output: 1 opened: 0\nALSA name: Midi Through Port-0 input: 1 output: 0 opened: 0\nALSA name: CH345 MIDI 1 input: 0 output: 1 opened: 0\nALSA name: CH345 MIDI 1 input: 1 output: 0 opened: 0\nDefault device: 0\n
*/